components/rpc/invoker/mosn/channel/httpchannel.go

```diff
@@ -70,36 +120,39 @@ func (h *httpChannel) Do(req *rpc.RPCRequest) (*rpc.RPCResponse, error) {
 	}
 
 	deadline, _ := ctx.Deadline()
-	if err = conn.SetDeadline(deadline); err != nil {
+	if err = conn.SetWriteDeadline(deadline); err != nil {
 		h.pool.Put(conn, true)
 		return nil, common.Error(common.UnavailebleCode, err.Error())
 	}
 
 	httpReq := h.constructReq(req)
 	defer fasthttp.ReleaseRequest(httpReq)
 
+	hstate := conn.state.(*hstate)
+	hstate.renew()
```

Comment:
why renew every time? we can do it only once on stateFunc

Reply:
ok